pytorch_lightning/trainer/training_loop.py

```diff
@@ -690,7 +694,7 @@ def run_training_batch(self, batch, batch_idx):
             signal=0,
             grad_norm_dic=grad_norm_dic,
             batch_log_metrics=batch_log_metrics,
-            training_step_output_for_epoch_end=opt_closure_result.training_step_output_for_epoch_end
+            training_step_output_for_epoch_end=all_training_step_output_for_epoch_end
```

Comment:
i believe the function that calls this method also needs to be fixed to take into account the list of outputs.

Reply:
`training_step_output_for_epoch_end=opt_closure_result.training_step_output_for_epoch_end`

this line only capture the last iteration outputs and discard the previous iterations.